silence/silence.go

```diff
@@ -469,6 +471,20 @@ func (s *Silences) GC() (int, error) {
 
 // ValidateMatcher runs validation on the matcher name, type, and pattern.
 var ValidateMatcher = func(m *pb.Matcher) error {
+	return validateClassicMatcher(m)
+}
+
+// InitFromFlags initializes the validation function from the flagger.
+func InitFromFlags(l log.Logger, f featurecontrol.Flagger) {
```

Comment:
I found an interesting issue here! If you create a silence with UTF-8 matchers and then restart Alertmanager with the "classic-matchers-parsing" feature flag, you can no longer edit or expire the silence because the validation function returns an error. However, the silence will still expire once it's expiration time has elapsed.

Reply:
I've attempted to fix that with this commit [here](https://github.com/prometheus/alertmanager/pull/3567/commits/c0dedde5e3360f55bf4a7d557e5ea9f1a9269a24).

Reply:
Great Find! I'll take a look once we have the other changes in.